### PR TITLE
docs: refresh readme instructions

### PR DESCRIPTION
**Pull request by cojenco**
Update instructions and links in quick start/ samples readme per @cbonnie friction log :)

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -34,20 +34,21 @@ In order to use this library, you first need to go through the following steps:
...
+`venv`_ is a tool to create isolated Python environments. The basic problem it
+addresses is one of dependencies and versions, and indirectly permissions.
 
-With `virtualenv`_, it's possible to install this library without needing system
+With `venv`_, it's possible to install this library without needing system
 install permissions, and without clashing with the installed system
 dependencies.
 
-.. _`virtualenv`: https://virtualenv.pypa.io/en/latest/
+.. _Set up a Python development environment: https://cloud.google.com/python/docs/setup
```
Comment (cbonnie) [inline README.rst:50]: Can we add a line that says something along the lines of: "Make sure you're using Python 3.3 and later, which includes venv by default."

File: README.rst [hunk 2]
```diff
@@ -68,30 +69,33 @@ Mac/Linux
...
 ~~~~~~~~~~~~~
 
 .. code:: python
 
+    # Imports the Google Cloud client library
     from google.cloud import storage
+
+    # Instantiates a client
     client = storage.Client()
+
```
Comment (cbonnie) [inline README.rst:97]: Nice comments!

File: samples/README.md [hunk 3]
```diff
@@ -21,7 +21,8 @@ Before running the samples, make sure you've followed the steps outlined in
 [Quick Start](https://github.com/googleapis/python-storage#quick-start).
 
 ### Authentication
-This sample requires you to have authentication setup. Refer to the [Authentication Getting Started Guide](https://cloud.google.com/docs/authentication/getting-started)
+The samples require you to have authentication setup. Provide authentication credentials to your application code by setting the environment variable `GOOGLE_APPLICATION_CREDENTIALS`.
+Refer to the [Authentication Set Up Guide](https://cloud.google.com/storage/docs/reference/libraries#setting_up_authentication)
 for instructions on setting up credentials for applications.
```
Comment (cbonnie) [inline samples/README.md:26]: I think just "Refer to the .... for more detailed instructions." would be fine.

File: README.rst [hunk 4]
```diff
@@ -68,30 +69,33 @@ Mac/Linux
...
+    # Creates a new bucket and uploads an object
     new_bucket = client.create_bucket('new-bucket-id')
     new_blob = new_bucket.blob('remote/path/storage.txt')
     new_blob.upload_from_filename(filename='/local/path.txt')
```
Comment (cbonnie) [inline README.rst:101]: Can we add a line to the very end of the quickstart doc? 

"Now that you've set up your Python client for Cloud Storage, you can get started running [Storage samples](https://github.com/googleapis/python-storage/blob/main/samples/README.md)." 

This just helps provide a next step.

**Review by cbonnie — COMMENTED** [review]
A couple nits, otherwise LGTM

**Review by cbonnie — APPROVED** [review]

File: README.rst (hunk 4 above)
Comment (cojenco) [reply README.rst:102]: Love the idea! Added a next step to the end of the quickstart.

File: README.rst (hunk 1 above)
Comment (cojenco) [reply README.rst:51]: Of course, done.

**Review by loferris — APPROVED** [review]

**Review by andrewsg — APPROVED** [review]
Good stuff! Thanks for working on this.

**Pull request merged**